src/test/java/uk/ac/cam/cl/dtg/isaac/api/AuthenticationFacadeIT.java

```diff
@@ -16,4 +53,58 @@
 
     // See E2E for logout test - response object is mocked in IT tests, preventing retrieval of transformed logout cookie
     // Usage of actual object may be possible in future but the complexity is currently prohibitive
+
+    @Test
+    public void resetPassword_emailRateLimits() throws InvalidKeySpecException, NoSuchAlgorithmException {
+        HttpSession mockSession = createNiceMock(HttpSession.class);
+        expect(mockSession.getAttribute(ANONYMOUS_USER)).andReturn(null).anyTimes();
+        expect(mockSession.getId()).andReturn("sessionIdEmail1").andReturn("sessionIdEmail2").andReturn("sessionIdEmail3");
+        replay(mockSession);
+        HttpServletRequest mockAuthRequest = createNiceMock(HttpServletRequest.class);
+        expect(mockAuthRequest.getHeader("X-Forwarded-For")).andReturn("0.0.0.0").anyTimes();
+        expect(mockAuthRequest.getSession()).andReturn(mockSession).anyTimes();
+        replay(mockAuthRequest);
+        HttpServletResponse  mockAuthResponse = createNiceMock(HttpServletResponse.class);
+
+        LocalAuthDTO targetUser = new LocalAuthDTO();
+        targetUser.setEmail("test-student@test.com");
+        targetUser.setPassword("123");
+
+        Response firstResetResponse = authenticationFacade.authenticateWithCredentials(mockAuthRequest, mockAuthResponse, AuthenticationProvider.SEGUE.toString(), targetUser);
+        assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), firstResetResponse.getStatus());
+
+        Response secondResetResponse = authenticationFacade.authenticateWithCredentials(mockAuthRequest, mockAuthResponse, AuthenticationProvider.SEGUE.toString(), targetUser);
+        assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), secondResetResponse.getStatus());
+
+        Response thirdResetResponse = authenticationFacade.authenticateWithCredentials(mockAuthRequest, mockAuthResponse, AuthenticationProvider.SEGUE.toString(), targetUser);
+        assertEquals(Response.Status.TOO_MANY_REQUESTS.getStatusCode(), thirdResetResponse.getStatus());
+    }
+
+    @Test
+    public void resetPassword_ipRateLimits() throws InvalidKeySpecException, NoSuchAlgorithmException {
+        HttpSession mockSession = createNiceMock(HttpSession.class);
+        expect(mockSession.getAttribute(ANONYMOUS_USER)).andReturn(null).anyTimes();
+        expect(mockSession.getId()).andReturn("sessionIdIp1").andReturn("sessionIdIp2").andReturn("sessionIdIp3");
+        replay(mockSession);
+        HttpServletRequest mockAuthRequest = createNiceMock(HttpServletRequest.class);
+        expect(mockAuthRequest.getHeader("X-Forwarded-For")).andReturn("0.0.0.0").anyTimes();
+        expect(mockAuthRequest.getSession()).andReturn(mockSession).anyTimes();
+        replay(mockAuthRequest);
+        HttpServletResponse  mockAuthResponse = createNiceMock(HttpServletResponse.class);
+
+        LocalAuthDTO targetUser = new LocalAuthDTO();
+        targetUser.setPassword("123");
```

Comment:
is there a config issue with the code scanning failing to detect that this is a test script?

Reply:
Possibly? I think that might need to be done in the git workflow rather than the codebase though. I had a quick look at the workflow file but I'm not sure what would need changing in there yet...

Reply:
can you raise a ticket to resolve that. We might not get it done any time soon but at least we'll have a record that it needs doing.